taggit/managers.py

```diff
@@ -221,13 +246,47 @@ def set(self, *tags):
 
     @require_instance_manager
     def remove(self, *tags):
-        self.through.objects.filter(**self._lookup_kwargs()).filter(
-            tag__name__in=tags).delete()
+        if not tags:
+            return
+
+        db = router.db_for_write(self.through, instance=self.instance)
+
+        qs = (self.through._default_manager.using(db)
```

Comment:
This introduces an additional SQL query, however, core's `ManyRelatedManager` also performs this query so I don't think it's unwarranted.